cmd/bucket_metadata_cache.go

```diff
@@ -0,0 +1,66 @@
+package cmd
+
+import (
+	"errors"
+	"sync"
+)
+
+var errCacheEntryNotFound = errors.New("Entry not found in cache")
+
+type BucketMetadataCache struct {
+	lock     sync.Mutex
+	metadata map[string]BucketMetadata
```

Comment:
As we will be calling 'Get()' in 99% of cases, can we use rwlock instead of mutex?